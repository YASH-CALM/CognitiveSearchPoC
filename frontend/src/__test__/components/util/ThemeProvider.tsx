import * as React from 'react'
import { Theme as UwpThemeProvider } from 'react-uwp/Theme';
import { theme } from '../../../components/App/theme';
export const withTheme = (children: string | JSX.Element) => <UwpThemeProvider theme={theme}>{children}</UwpThemeProvider>