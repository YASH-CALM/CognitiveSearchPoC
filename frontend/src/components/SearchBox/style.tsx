export const searchBoxStyle: React.CSSProperties = {
    margin: '0 auto',
    width: '80%',
}

export const textBoxStyle: React.CSSProperties = {
    background: '#00000000',
    color: "#EEEEEE88",
    height: 36,
    paddingLeft: 16,
    paddingRight: 36,
    width: '100%',
}

export const searchIconStyle: React.CSSProperties = {
    height: 40,
    marginBottom: 9,
    marginRight: -36,
    paddingBottom: 9,
    width: 36,
}
