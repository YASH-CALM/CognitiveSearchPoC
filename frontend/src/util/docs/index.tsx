import { IDocument } from 'src/types/documents';

export const rNumberRange = (max: number, min?: number): number => {
    return Math.floor(Math.random() * max) + (min === 0 ? 0 : (min || 1));
}

export const rLetter = (): string => {
    return 'abcdefghijklmnopqrstuvwxyz'[rNumberRange(26, 0)];
}

const loremIpsum: string[] = `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
Integer sed molestie quam, eleifend imperdiet turpis.
In ut mauris enim. Curabitur eget tempus leo. Etiam
faucibus placerat vulputate. Suspendisse potenti.
In risus velit, scelerisque maximus vulputate vitae,
mollis sed mi. In eleifend, tortor in ultricies pulvinar,
lorem ipsum tincidunt eros, ut dignissim orci ipsum ut
tortor. Mauris porta lorem non nunc malesuada, sed euismod
metus gravida. Morbi tempus ante ac cursus finibus.
Phasellus elementum ac justo ut rutrum. Aliquam id tempus
tellus, vitae consectetur leo. Mauris sapien urna, feugiat
sed nulla id, dictum pharetra velit. Quisque accumsan
blandit orci nec sollicitudin. Phasellus vel ipsum consequat
arcu finibus pretium nec ut leo.`.split(' ');

export const rTag = (): string => loremIpsum[rNumberRange(loremIpsum.length-1, 0)];

export const rDocument = (): IDocument => ({
    pages: Array(rNumberRange(5)).fill(0).map(m => rNumberRange(5)).map(p => ({
        source: `/docs/phb/${p}.png`,
        tags: Array(rNumberRange(26)).fill(0).map(rTag),
    })),
    title: `${rLetter()}${rLetter()}${rLetter()}_${rNumberRange(1000000000)}`,
})

export const docs = (): IDocument[] => Array(rNumberRange(50,25)).fill(0).map(rDocument);