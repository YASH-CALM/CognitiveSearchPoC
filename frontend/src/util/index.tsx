export const isArrayEmpty = (array?: any[]): boolean => !array || !array.length;
export const isDefinedNumber = (n?: number) : boolean => n != null;