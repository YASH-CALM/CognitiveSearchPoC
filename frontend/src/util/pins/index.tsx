import { IInfoboxWithPushPin } from 'src/types/pins';

export const pins: IInfoboxWithPushPin[] = [
    {
        infobox: {
            description: 'this location was entered as a longitude latitude pair "[36, -112]"',
            title: 'Grand Canyon',
        },
        location: [36, -112],
        pushPin: {
            description: 'Grand Canyon',
            options: { color: 'orange' },
            title: 'Grand Canyon',
        }
    },
    {
        infobox: {
            description: 'this location was entered as an address "841 Avenue of the Cities"',
            title: 'YASH Office',
        },
        location: ['841 Avenue of the Cities'],
        pushPin: {
            description: 'Yash Office',
            options: { color: '#005ba1' },
            title: 'YASH',
        }
    },
    {
        infobox: {
            description: 'this location was entered as a longitude latitude pair "[45, -84]"',
            title: 'Lake Superior',
        },
        location: [45, -84],
        pushPin: {
            description: 'Lake Superior',
            options: { color: 'lightgreen' },
            title: 'Lake Superior',
        }
    },
    {
        infobox: {
            description: 'this location was entered as the name of the location "Statue of Liberty"',
            title: 'Statue of Liberty',
        },
        location: ['Statue of Liberty'],
        pushPin: {
            description: 'Statue of Liberty',
            options: { color: 'grey' },
            title: 'Statue of Liberty',
        }
    },
]