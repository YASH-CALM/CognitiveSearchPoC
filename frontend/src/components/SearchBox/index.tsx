import * as React from 'react';
import IconButton from 'react-uwp/IconButton';
import TextBox from 'react-uwp/TextBox';
import * as Style from './style';

interface ISearchBoxProps {
    placeholder?: string,
    value: string,
    onChange: (e: any) => void,
    searchAction: (q: string) => void,
}

const SearchBox: React.SFC<ISearchBoxProps> = ({placeholder, value, onChange, searchAction}) => {
    const onClick=() => searchAction(value);
  return <div style={Style.searchBoxStyle}>
      <TextBox
        defaultValue={value}
        onChange={onChange} 
        placeholder={placeholder}
        style={Style.textBoxStyle}
      >
          <IconButton onClick={onClick} style={Style.searchIconStyle}>Search</IconButton>
      </TextBox>
  </div>;
};

export default SearchBox;