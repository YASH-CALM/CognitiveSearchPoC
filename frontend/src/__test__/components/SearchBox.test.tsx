import * as React from 'react'
import { createRenderer as ShallowRenderer } from 'react-test-renderer/shallow'
import SearchBox from '../../components/SearchBox'
import { withTheme } from './util/ThemeProvider'

describe('SearchBox Component', () => {
    const renderer = ShallowRenderer()
    const props = {
        onChange: jest.fn(),
        searchAction: jest.fn(),
        value: 'value',
    }
    it('should match snapshot', () => {
        renderer.render(withTheme(<SearchBox {...props}/>))
        const result = renderer.getRenderOutput()
        expect(result).toMatchSnapshot()
    })
})