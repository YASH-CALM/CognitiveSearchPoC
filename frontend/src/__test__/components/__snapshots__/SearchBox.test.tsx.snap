// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component should match snapshot 1`] = `
<div
  className="theme-root-2366926892 react-uwp-theme-dark"
>
  <RenderToBody
    className="fluent-background-3499734374"
  />
  <RenderToBody>
    <ToastWrapper />
  </RenderToBody>
  <SearchBox
    onChange={[MockFunction]}
    searchAction={[MockFunction]}
    value="value"
  />
</div>
`;
